edx_sga/utils.py

```diff
@@ -9,10 +9,33 @@
 
 import pytz
 from django.conf import settings
-from django.core.files.storage import default_storage
+from django.core.files.storage import default_storage as django_default_storage, get_storage_class
 from edx_sga.constants import BLOCK_SIZE
 
 
+def get_default_storage():
+    """
+    Get config for storage from settings, use DJango's default_storage if no such settings defined
+    """
+    # .. setting_name: SGA_STORAGE_SETTINGS
+    # .. setting_default: {}
+    # .. setting_description: Specifies the storage class and keyword arguments to use in the constructor
+    # .. setting_example: {
+    #        STORAGE_CLASS: 'storage',
+    #        STORAGE_KWARGS: {}
+    #    }
+    sga_storage_settings = getattr(settings, "SGA_STORAGE_SETTINGS", None)
+
+    if sga_storage_settings:
+        return get_storage_class(
```

Comment:
I have two things in mind regarding this:

1. nit: [get_storage_class in Django](https://docs.djangoproject.com/en/4.1/ref/files/storage/#django.core.files.storage.get_storage_class) automatically fallbacks to and returns default storage, e.g. `FileStorage` if the provided args are `None or empty`. We can use this and easily remove some condition statements and code lines.

    Let me know, If you think that would be less readable.
2. nit: This can throw `ImportError` if the provided `settings/class name` isn't correct.
    - Now we can either 1) let the error bubble up in the logs or 2) Catch it and raise a properly formatted named exception such as ` ImproperlyConfigured from django.core.exceptions`. I'm fine either way so this change is kind of optional, I just wanted to mention it anyway.

Reply:
1. Yes, but we still have to call the constructor with the keyword arguments too. It would still require the conditional statement.

2. Yes it can throw an error, but there is a single source from which this value will come, that's why skipped throw catch in first place.